plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -131,16 +130,11 @@ private static Optional<String> toPredicate(PinotColumnHandle pinotColumnHandle,
         String predicateArgument = pinotColumnHandle.isAggregate() ? pinotColumnHandle.getExpression() : quoteIdentifier(pinotColumnHandle.getColumnName());
         ValueSet valueSet = domain.getValues();
         if (valueSet.isNone()) {
-            verify(!domain.isNullAllowed(), "IS NULL is not supported due to different null handling semantics. See https://docs.pinot.apache.org/developers/advanced/null-value-support");
+            if (domain.isNullAllowed()) {
```

Comment:
Should this be done only if the table has null handling enabled in the config? How would it know the difference between `'null', Long.MIN_VALUE`, etc. vs a null value? Is that handled in v1.0.0?